python/python/bystro/proteomics/listener_annotation_interface.py

```diff
@@ -0,0 +1,122 @@
+import argparse
+
+import logging
+from ruamel.yaml import YAML
+from pathlib import Path
+import pyarrow.dataset as ds
+from opensearchpy import OpenSearch
+
+from bystro.beanstalkd.worker import listen, QueueConf, ProgressPublisher
+from bystro.beanstalkd.messages import BaseMessage, CompletedJobMessage, SubmittedJobMessage
+from bystro.proteomics.annotation_interface import (
+    get_annotation_result_from_query,
+    join_annotation_result_to_proteomics_dataset,
+)
+
+logger = logging.getLogger(__file__)
+
+EFS_BASE_PATH = "/seqant/user-data/"
+PROTEOMICS_TUBE = "proteomics"
+
+
+class ProteomicsJobData(BaseMessage, frozen=True, rename="camel"):
+    """
+    The expected JSON message for the Proteomics job.
+
+    Parameters
+    ----------
+    submission_id: str
+        The unique identifier for the job.
+    data_path: str
+        The path to the proteomics data file.
+    out_dir: str
+        The directory to write the results to.
+    annotation_query: str
+        The query string used to fetch annotation data from OpenSearch index.
+    index_name: str
+        The name of the OpenSearch index from which annotation data will be queried.
+    """
+
+    submission_id: str
```

Comment:
You don't need to provide submission_id here, as BaseMessage already provides that